### PR TITLE
feat: support postgres wire protocol

### LINKED ISSUE #44: Support PostgreSQL wire protocol
### Describe This Problem

MySQL support is already fixed in #26, it would be great if PostgreSQL is also supported.

### Proposal

Implement database trait for PostgreSQL
- https://docs.rs/postgres/latest/postgres/struct.Client.html

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Lilit0x**
# Which issue does this PR close?

Closes #44 

# What changes are included in this PR?
Adding support for PostgreSQL databases to run tests against

# Are there any user-facing changes?
- Multiple options for the `type` argument
- 
# How does this change test
CI

### CODE REVIEW

**Comment by Lilit0x** [comment]
@jiacai2050, I have the basic outline here, but I am quite stuck on how to integrate it with CLI through the `EnvController` seeing that it requires a `DB` type. And now, there are two types of `DB`. 
So, I just repicated the `EnvController` trait for Postgres too and I know that's not the way to go, but I guess it is a starting point.

File: sqlness/src/database_impl/postgresql.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+use async_trait::async_trait;
+use postgres::{Client, Config, NoTls, Row};
+use std::{
+    fmt::Display,
+    sync::{Arc, Mutex},
+};
+
+use crate::{Database, DatabaseConfig, QueryContext};
+
+/// DatabaseBuilder for PostgreSQL
```
Comment (jiacai2050) [inline sqlness/src/database_impl/postgresql.rs]: This is not a builder type, I suggest remove this comments.

File: sqlness-cli/src/main.rs [hunk 2]
```diff
@@ -65,6 +66,21 @@ impl EnvController for CliController {
     async fn stop(&self, _env: &str, _db: Self::DB) {}
 }
 
+struct PostgresController {
+    db_config: DatabaseConfig,
+}
+
+#[async_trait]
+impl EnvController for PostgresController {
```
Comment (jiacai2050) [inline sqlness-cli/src/main.rs]: I think add PostgresController is a valid option, and you need to also update L80 to something like this:
```
let ctrl = match type {
mysql => ...
postgres => ...
}
```

**Comment by tanruixiang** [comment]
@Lilit0x  Hi, I've helped you make the appropriate changes based on the REVIEW, if you have time, please help with the REVIEW.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**